packages/console/src/hooks/use-current-subscription-plan.ts

```diff
@@ -0,0 +1,24 @@
+import { useMemo } from 'react';
+
+import useCurrentSubscription from './use-current-subscription';
+import useSubscriptionPlans from './use-subscription-plans';
+
+const useCurrentSubscriptionPlan = () => {
+  const { data: subscription, error: fetchSubscriptionError } = useCurrentSubscription();
+  const { data: subscriptionPlans, error: fetchSubscriptionPlansError } = useSubscriptionPlans();
+
+  const isLoadingSubscription = !subscription && !fetchSubscriptionError;
+  const isLoadingSubscriptionPlans = !subscriptionPlans && !fetchSubscriptionPlansError;
+
+  const currentPlan = useMemo(
+    () => subscriptionPlans?.find(({ id: planId }) => planId === subscription?.planId),
+    [subscription?.planId, subscriptionPlans]
+  );
+
+  return {
+    currentPlan,
+    isLoading: isLoadingSubscription || isLoadingSubscriptionPlans,
+  };
+};
```

Comment:
Can we make the return type similar to a `useSWR`?

```ts
return { data: currentPlan, error: fetchSubscriptionError ?? fetchSubscriptionPlansError };
```

Reply:
Cool, updated.